### PR TITLE
Format2 work211117

### PR DESCRIPTION
**Pull request by ava-fred**
This PR is a rebase of PR #170. It contains a new implementation of the Format language using new mwe2 fragments.

### CODE REVIEW

File: com.avaloq.tools.ddk.xtext.format.generator/src/com/avaloq/tools/ddk/xtext/format/generator/FormatFragment2.xtend [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    .addTypeToType(INodeModelFormatter.typeRef, RegionNodeModelFormatter.typeRef)
+    .addTypeToType(INodeModelStreamer.typeRef, DirectNodeModelStreamer.typeRef)
+    .contributeTo(language.runtimeGenModule)
+    if (projectConfig.runtime.manifest !== null) {
+      projectConfig.runtime.manifest.requiredBundles += "org.eclipse.emf.ecore"
+      projectConfig.runtime.manifest.requiredBundles += RUNTIME_PLUGIN
+    }
+    if (projectConfig.eclipsePlugin.manifest !== null) {
+      projectConfig.eclipsePlugin.manifest.requiredBundles += RUNTIME_PLUGIN
+    }
```
[the remark this answers is not in the document]
Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format.generator/src/com/avaloq/tools/ddk/xtext/format/generator/FormatFragment2.xtend:106]: done

File: com.avaloq.tools.ddk.xtext.format/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -12,18 +12,20 @@ Require-Bundle: org.eclipse.xtext,
  org.eclipse.emf.codegen.ecore;resolution:=optional,
  org.eclipse.emf.mwe.utils;resolution:=optional,
  org.eclipse.emf.mwe2.launch;resolution:=optional,
+ org.eclipse.equinox.common;resolution:=optional,
  com.avaloq.tools.ddk.xtext,
  org.eclipse.xtext.util,
  org.eclipse.emf.ecore,
  org.eclipse.emf.common,
- org.antlr.runtime,
+ org.antlr.runtime;bundle-version="[3.2.0,3.2.1)",
```
[the remark this answers is not in the document]
Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format/META-INF/MANIFEST.MF:20]: done

File: com.avaloq.tools.ddk.xtext.format/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -12,18 +12,20 @@ Require-Bundle: org.eclipse.xtext,
...
  org.eclipse.osgi,
  org.eclipse.core.runtime,
  org.eclipse.core.resources,
+ org.eclipse.xtend.lib,
  org.eclipse.xtext.common.types,
  org.eclipse.xtext.xbase,
  org.eclipse.xtext.xbase.lib,
- org.objectweb.asm;resolution:=optional
+ org.objectweb.asm;bundle-version="[9.1.0,9.2.0)";resolution:=optional
```
[the remark this answers is not in the document]
Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format/META-INF/MANIFEST.MF:28]: done

File: com.avaloq.tools.ddk.xtext.format.ui/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -12,7 +12,7 @@ Require-Bundle: com.avaloq.tools.ddk.xtext.format;visibility:=reexport,
  org.eclipse.xtext.ui.shared,
  org.eclipse.ui,
  org.eclipse.xtext.builder,
- org.antlr.runtime,
+ org.antlr.runtime;bundle-version="[3.2.0,3.2.1)",
```
[the remark this answers is not in the document]
Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format.ui/META-INF/MANIFEST.MF:15]: done

File: com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore [hunk 5]
```diff
@@ -3,7 +3,7 @@
     xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore" name="format" nsURI="http://www.avaloq.com/tools/ddk/xtext/format/Format"
     nsPrefix="format">
   <eClassifiers xsi:type="ecore:EClass" name="FormatConfiguration">
-    <eStructuralFeatures xsi:type="ecore:EReference" name="targetGrammar" eType="ecore:EClass platform:/plugin/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//Grammar"/>
+    <eStructuralFeatures xsi:type="ecore:EReference" name="targetGrammar" eType="ecore:EClass platform:/resource/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//Grammar"/>
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore:6]: It looks we still have this one left. Could you revert it as well?

File: com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore [hunk 6]
```diff
@@ -31,7 +31,7 @@
     <eStructuralFeatures xsi:type="ecore:EAttribute" name="override" eType="ecore:EDataType platform:/resource/org.eclipse.emf.ecore/model/Ecore.ecore#//EBoolean"/>
   </eClassifiers>
   <eClassifiers xsi:type="ecore:EClass" name="GrammarRule" eSuperTypes="#//Rule">
-    <eStructuralFeatures xsi:type="ecore:EReference" name="targetRule" eType="ecore:EClass platform:/plugin/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//AbstractRule"/>
+    <eStructuralFeatures xsi:type="ecore:EReference" name="targetRule" eType="ecore:EClass platform:/resource/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//AbstractRule"/>
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore:34]: revert?

File: com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore [hunk 7]
```diff
@@ -43,14 +43,14 @@
...
-    <eStructuralFeatures xsi:type="ecore:EReference" name="rule" eType="ecore:EClass platform:/plugin/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//AbstractRule"/>
-    <eStructuralFeatures xsi:type="ecore:EReference" name="keyword" eType="ecore:EClass platform:/plugin/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//Keyword"/>
+    <eStructuralFeatures xsi:type="ecore:EReference" name="assignment" eType="ecore:EClass platform:/resource/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//Assignment"/>
+    <eStructuralFeatures xsi:type="ecore:EReference" name="ruleCall" eType="ecore:EClass platform:/resource/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//RuleCall"/>
+    <eStructuralFeatures xsi:type="ecore:EReference" name="self" eType="ecore:EClass platform:/resource/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//AbstractRule"/>
+    <eStructuralFeatures xsi:type="ecore:EReference" name="rule" eType="ecore:EClass platform:/resource/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//AbstractRule"/>
+    <eStructuralFeatures xsi:type="ecore:EReference" name="keyword" eType="ecore:EClass platform:/resource/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//Keyword"/>
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore:50]: revert?

File: com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore [hunk 8]
```diff
@@ -43,14 +43,14 @@
...
   </eClassifiers>
   <eClassifiers xsi:type="ecore:EClass" name="GrammarElementLookup">
-    <eStructuralFeatures xsi:type="ecore:EReference" name="rule" eType="ecore:EClass platform:/plugin/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//AbstractRule"/>
+    <eStructuralFeatures xsi:type="ecore:EReference" name="rule" eType="ecore:EClass platform:/resource/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//AbstractRule"/>
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore:53]: revert?

File: com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore [hunk 9]
```diff
@@ -70,7 +70,7 @@
         eType="#//Matcher" containment="true"/>
   </eClassifiers>
   <eClassifiers xsi:type="ecore:EClass" name="GroupBlock">
-    <eStructuralFeatures xsi:type="ecore:EReference" name="grammarElement" eType="ecore:EClass platform:/plugin/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//CompoundElement"/>
+    <eStructuralFeatures xsi:type="ecore:EReference" name="grammarElement" eType="ecore:EClass platform:/resource/org.eclipse.xtext/org/eclipse/xtext/Xtext.ecore#//CompoundElement"/>
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore:73]: revert?

**Review by rubenporras — COMMENTED** [review]
Could we do the remaining comments?

File: com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore (hunk 5 above)
Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore]: done

Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore]: done

File: com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore (hunk 6 above)
Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore]: done

File: com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore (hunk 7 above)
Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore]: done

Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore]: done

Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore]: done

File: com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore (hunk 9 above)
Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore]: done

File: com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore (hunk 6 above)
Comment (ava-fred) [reply com.avaloq.tools.ddk.xtext.format/model/generated/Format.ecore]: done

File: com.avaloq.tools.ddk.xtext.format.ui/src-gen/com/avaloq/tools/ddk/xtext/format/ide/contentassist/antlr/PartialFormatContentAssistParser.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
+/*
+ * generated by Xtext 2.25.0
+ */
+package com.avaloq.tools.ddk.xtext.format.ide.contentassist.antlr;
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext.format.ui/src-gen/com/avaloq/tools/ddk/xtext/format/ide/contentassist/antlr/PartialFormatContentAssistParser.java:4]: I have noticed now that this commit introduces '.ide' package. We need to move that to its own plug-in, so that it can be used on LSP without pulling the full UI one, but we can do this on another commit.

**Review by rubenporras — APPROVED** [review]
